src/identities/mnemonic.ts

```diff
@@ -1,30 +1,177 @@
+    this.masterPrivateKeyNode = bip32fromSeed(this.walletSeed, this.network);
+    // generate the master blinding key from the seed
+    this.masterBlindingKeyNode = slip77fromSeed(this.walletSeed);
   }
 
-  signPset(psetBase64: string): string {
-    console.log(psetBase64);
-    return '';
+  private getNextKeypair(): { publicKey: Buffer; privateKey: Buffer } {
+    const baseNode = this.masterPrivateKeyNode.derivePath(this.derivationPath);
+    const wif: string = baseNode.deriveHardened(this.index).toWIF();
+    const { publicKey, privateKey } = ECPair.fromWIF(wif, this.network);
+    this.index += 1;
+    return { publicKey: publicKey!, privateKey: privateKey! };
+  }
+
+  private getBlindingKeyPair(
+    scriptPubKey: Buffer
+  ): { publicKey: Buffer; privateKey: Buffer } {
+    const { publicKey, privateKey } = this.masterBlindingKeyNode.derive(
+      scriptPubKey
+    );
+    return { publicKey: publicKey!, privateKey: privateKey! };
+  }
+
+  getNextConfidentialAddress(): AddressInterfaceExtended {
+    const currentIndex = this.index;
+    // get the next key pair
+    const keyPair = this.getNextKeypair();
+    // use the public key to compute the scriptPubKey
+    const script: Buffer = payments.p2wpkh({
+      pubkey: keyPair.publicKey!,
+      network: this.network,
+    }).output!;
+    // generate the blindKeyPair from the scriptPubKey
+    const blindingKeyPair = this.getBlindingKeyPair(script);
+    // with blindingPublicKey & signingPublicKey, generate the confidential address
+    const { confidentialAddress } = payments.p2wpkh({
+      pubkey: keyPair.publicKey!,
+      blindkey: blindingKeyPair.publicKey!,
+      network: this.network,
+    });
+    // create the address generation object
+    const newAddressGeneration: AddressInterfaceExtended = {
+      address: {
+        confidentialAddress: confidentialAddress!,
+        blindingPrivateKey: blindingKeyPair.privateKey!.toString('hex'),
+      },
+      derivationPath: `${this.derivationPath}/${currentIndex}`,
+      signingPrivateKey: keyPair.privateKey!.toString('hex'),
+      scriptPubKey: script,
+    };
+    // store the generation inside local cache
+    this.confidentialAddresses.push(newAddressGeneration);
+    // return the generation data
+    return newAddressGeneration;
+  }
+
+  unblindUtxo(blindedUtxo: Output): UnblindOutputResult {
+    const blindingPrivKey: Buffer = this.masterBlindingKeyNode.derive(
+      blindedUtxo.script
+    ).privateKey!;
+
+    return confidential.unblindOutput(
+      blindedUtxo.nonce,
+      blindingPrivKey,
+      blindedUtxo.rangeProof!,
+      blindedUtxo.value,
+      blindedUtxo.asset,
+      blindedUtxo.script
+    );
+  }
+
+  async signPset(psetBase64: string): Promise<string> {
+    const pset = Psbt.fromBase64(psetBase64);
+    const signInputPromises: Array<Promise<void>> = [];
+    const findInAddresses = (script: Buffer) =>
+      this.confidentialAddresses.find(addr => addr.scriptPubKey.equals(script));
```

Comment:
indeed, as per previous comment I guess we should store via key, the script would be very handy, instead of looping all the array.